testing/matrix-sdk-test/src/event_factory.rs

```diff
@@ -871,9 +873,11 @@ impl EventFactory {
         &self,
         creator_user_id: &UserId,
         room_version: RoomVersionId,
+        room_type: Option<RoomType>,
```

Comment:
Instead of adding this argument here, we should add a new method here:

https://github.com/matrix-org/matrix-rust-sdk/blob/3aeface647183345973e310930fa037bc603db04/testing/matrix-sdk-test/src/event_factory.rs#L453-L465

This removes the need to modify *every* call site of this method which in turn makes the PR smaller. This change could have been a separate PR as well.

Reply:
Oh this is neat, didn't realise it aws a thing! 👍